networking.md

```diff
@@ -0,0 +1,463 @@
+# Networking
+
+## Architecture
+
+![Network diagram](images/network_diagram.jpg)
+
+### Ingress
+
+#### Public endpoints
+Per default public endpoints get created as routes.
+
+The OCP route hostname gets registered as the keystone public endpoint for the service.
+```bash
++--------------+-----------+------------------------------------------------------------------+
+| Service Name | Interface | URL                                                              |
++--------------+-----------+------------------------------------------------------------------+
+| <service     | public    | http://<service>-public-openstack.apps.ostest.test.metalkube.org |
++--------------+-----------+------------------------------------------------------------------+
+```
+
+#### VIP endpoints
+MetalLB service VIP (currently - in layer 2 mode) used for endpoints on networks which compute nodes need to be able to access. The connection is terminated at one node at the metallb VIP. From there spread to all service pods via the networking rules kube-proxy has set.
+
+The OCP LoadBalancer type service name gets registered as the keystone endpoint for the service. Controlplane services running on OCP will resolve those to the ClusterIP of the service, the Compute nodes have to resolve it to the External-IP of the LoadBalancer.
+```bash
++--------------+-----------+---------------------------------------------------------------+
+| Service Name | Interface | URL                                                           |
++--------------+-----------+---------------------------------------------------------------+
+| <service>    | internal  | http://<service>-internal.openstack.svc:9292                  |
++--------------+-----------+---------------------------------------------------------------+
+```
+
+#### Individual endpoints
+For services like OVS access via static IP on the isolated network is required.
+
+This is right now a TODO and targeted when the IPAM solution, which is also required for EDPM nodes has finalized. Right now when the OVS pods restart, they'll likely get different IPs.
+
+The goal is for those pods to use static IPAM plugin and pass the pod IP via the annotation.
+```yaml
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack", "ips": ["172.17.0.20/24"]}]'
+```
+
+### Egress
+For outgoing isolated network traffic, pods be attached to additional networks using multus. The operators add the `k8s.v1.cni.cncf.io/networks` annotation  with referencing the list of network-attachment-definitios the deployemtn pods should be attached:
```

Comment:
`s/pods be/pods are`

I would also rewrite the second sentence to something like:

```
The operators add `k8s.v1.cni.cncf.io/networks` annotations that reference the list of network-attachment-definitions to which the deployment pods should be attached:
```